src/Turbo/src/Twig/TwigExtension.php

```diff
@@ -50,6 +56,24 @@ public function turboStreamListen(Environment $env, $topic, ?string $transport =
             $topic = new TopicSet($topic);
         }
 
-        return $this->turboStreamListenRenderers->get($transport)->renderTurboStreamListen($env, $topic);
+        if (
+            null !== $this->authorization
+            && null !== $this->requestStack
+            && (isset($options['subscribe']) || isset($options['publish']) || isset($options['additionalClaims']))
+            // @phpstan-ignore-next-line
+            && null !== $request = method_exists($this->requestStack, 'getMainRequest') ? $this->requestStack->getMainRequest() : $this->requestStack->getMasterRequest()
+        ) {
+            $this->authorization->setCookie(
+                $request,
+                $options['subscribe'] ?? [],
+                $options['publish'] ?? [],
+                $options['additionalClaims'] ?? [],
+                $transport,
+            );
+
+            unset($options['subscribe'], $options['publish'], $options['additionalClaims']);
+        }
+
+        return $this->turboStreamListenRenderers->get($transport)->renderTurboStreamListen($env, $topic, $options);
```

Comment:
You cannot pass options as third parameter, as the interface only declare "options" in docblock.

Reply:
I'm not really sure how to do that, then ?